File: frontend/src/api/system.ts
import { apiGet } from './client';

export interface SystemMonitor {
  cpu_percent: number;
  ram_percent: number;
  gpu: {
    name: string;
    util_percent: number;
    mem_used_mb: number;
    mem_total_mb: number;
  }[];
}

export function getHealth(): Promise<{ status: string }> {
  return apiGet<{ status: string }>('/health');
}

export function getMonitor(): Promise<SystemMonitor> {
  return apiGet<SystemMonitor>('/system/monitor');
}

File: src/adapters/input/web/schemas/system.py
from __future__ import annotations

from pydantic import BaseModel


class GPUInfo(BaseModel):
    name: str
    util_percent: float | None = None
    mem_used_mb: float | None = None
    mem_total_mb: float | None = None


class SystemMonitorResponse(BaseModel):
    cpu_percent: float
    ram_percent: float
    gpu: list[GPUInfo] = []


class HealthResponse(BaseModel):
    status: str = "ok"
    version: str = "1.0"
